src/alchemlyb/estimators/ti_gaussian_quadrature_.py

```diff
@@ -0,0 +1,228 @@
+        # variance calculation assumes no correlation between points
+        # used to calculate mean
+        self.means = dHdl.groupby(level=dHdl.index.names[1:]).mean()
+        self.variances = np.square(dHdl.groupby(level=dHdl.index.names[1:]).sem())
+        # suggested lambda values and corresponding weights 
+        special_points = self.get_quadrature_points()
+        weights = []
+        # check if the lambdas in the simulations match the suggested values
+        lambda_list, means_list, variances_list = self.separate_dhdl()
+        for lambdas in lambda_list:
+            num_lambdas = len(lambdas)
+            if (num_lambdas not in special_points) or (not np.allclose(lambdas, special_points[num_lambdas][0], rtol=0.1)):
+                raise ValueError('The lambda values for gaussian quadrature are not supported, please use trapezoidal rule instead.')
+            weights.extend(special_points[num_lambdas][1])
+        # means_new and variances_new are similar to means and variances, but with only values relevant to each lambda type (for multi-lambda situation)
+        means_new = concat(means_list)
+        mean_values = means_new.to_numpy()
+        variances_new = concat(variances_list)
+        variance_values = variances_new.to_numpy()
+
+        # apply gaussian quadrature multiplication at each lambda state 
+        deltas = weights * mean_values
+        d_deltas_squared = np.square(weights) * variance_values
+        # build matrix of deltas between each state
+        adelta = np.zeros((len(deltas), len(deltas)))
+        ad_delta = np.zeros_like(adelta)
+
+        for j in range(len(deltas)):
+            out = []
+            dout = []
+            for i in range(len(deltas) - j):
+                # Append cumulative free energy value from state i to i+j
+                out.append(deltas[i] + deltas[i + 1 : i + j + 1].sum())
+                # Append cumulative squared deviation of free energy from state i to i+j
+                dout.append(d_deltas_squared[i] + d_deltas_squared[i + 1 : i + j + 1].sum())
+            
+            adelta += np.diagflat(np.array(out), k=j)
+            ad_delta += np.diagflat(np.array(dout), k=j)
+
+        # yield standard delta_f_ cumulative free energies from one state to another
+        self._delta_f_ = pd.DataFrame(
+            adelta, columns=means_new.index.values, index=means_new.index.values
+        )
+        
+        # yield standard deviation d_delta_f_ between each state
+        self._d_delta_f_ = pd.DataFrame(
+            np.sqrt(ad_delta),
+            columns=variances_new.index.values,
+            index=variances_new.index.values,
+        )
+        
+        self.dhdl = self.means
+        self.dhdl.attrs = dHdl.attrs
+        self._states_ = means_new.index.values.tolist()
+        self._delta_f_.attrs = dHdl.attrs
+        self._d_delta_f_.attrs = dHdl.attrs
+
+        return self
+
+    def separate_dhdl(self):
```

Comment:
Do you need this as a public function? We have `TI.separate_dhdl()` so we would like this to have a different name.

Reply:
Might make this into a static function that takes `self.means` as input and return the results.

Reply:
I have changed this to a static function with a different name and it takes means and variances directly.